src/actions.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * action types
+ */
+
+export const SET_VERSION = 'SET_VERSION';
+export const UPDATE_VERSION_INPUT = 'UPDATE_VERSION_INPUT';
+export const SUBMIT_VERSION = 'SUBMIT_VERSION';
+export const UPDATE_LATEST_CHANNEL_VERSIONS = 'UPDATE_LATEST_CHANNEL_VERSIONS';
+export const UPDATE_STATUSES = 'UPDATE_STATUSES';
+
+/*
+ * other constants
+ */
+
+export const Channels = {
+  NIGHTLY: 'Nightly',
+  BETA: 'Beta',
+  RELEASE: 'Release',
+  ESR: 'ESR',
+};
+
+export const Statuses = {
+  ERROR: 'error',
+  EXISTS: 'exists',
+  INCOMPLETE: 'incomplete',
+  MISSING: 'missing',
+};
+
+/*
+ * action creators
+ */
+
+export function setVersion(version) {
+  return {type: SET_VERSION, version};
+}
+
+export function updateVersionInput(version) {
+  return {type: UPDATE_VERSION_INPUT, version};
+}
+
+export function submitVersion() {
+  return {type: SUBMIT_VERSION};
+}
+
+export function updateLatestChannelVersions(versions) {
+  return {type: UPDATE_LATEST_CHANNEL_VERSIONS, versions};
+}
+
+export function updateStatus(statuses) {
+  return {type: UPDATE_STATUSES, statuses};
+}
+
+// ASYNC (THUNK) ACTIONS.
+
+// Fetching the statuses.
+function fetchStatus(version) {
+  const stateToUrl = {
+    archive: 'archive',
+    release_notes: 'bedrock/release-notes',
+    security_advisories: 'bedrock/security-advisories',
+    download_links: 'bedrock/download-links',
+    product_details: 'product-details',
+  };
+  return Promise.all(
+    Object.keys(stateToUrl).map(key => {
+      const endpoint = stateToUrl[key];
+      return fetch(
+        `https://pollbot.dev.mozaws.net/v1/firefox/${version}/${endpoint}`,
+      )
+        .then(resp => resp.json())
+        .then(details => ({key, details}));
+    }),
+  ).then(results =>
+    results.reduce((acc, {key, details}) => {
+      acc[key] = details;
+      return acc;
+    }, {}),
+  );
+}
+
+export function requestStatus(version) {
+  const notifyChanges = changed => {
+    if (Notification.permission === 'granted') {
+      const names = changed.map(s => s.replace('_', ' ')).join(', ');
+      new Notification(`${document.title}: Status of ${names} changed.`);
+    }
+  };
```

Comment:
I guess there's no point in having an action/reducer for this because it's outside the bounds of our glorious redux-managed state, because it's a stateful web API?

Reply:
I guess... do you think it would be better if we stored the notification permission in the store?

Reply:
No, I think this is correct, just wanted to make sure I understood why.